### ISSUE TITLE
version of libraries in `requirements.txt`

### ISSUE THREAD

**Issue by modenaxe** [antoinefalisse/predictsim_mtp#4]
Hello, I followed the README instructions and I was trying to run the `main.py` script in the repository but I run in the following error:

```
Traceback (most recent call last):
  File "C:\Users\lucam\Desktop\predictsim_mtp\main.py", line 741, in <module>
    opti.set_initial(a_col, gACol.to_numpy().T)
  File "D:\Programs\python3\Miniconda\envs\3dpredsim\lib\site-packages\casadi\casadi.py", line 27519, in set_initial
    return _casadi.Opti_set_initial(self, *args)
NotImplementedError: Wrong number or type of arguments for overloaded function 'Opti_set_initial'.
  Possible prototypes are:
    set_initial(self,[MX])
    set_initial(self,MX,DM)
  You have: '(Opti,MX,np.array(float))'
```
Before trying to debug in casadi, however, I wanted to be sure that I am using the same version of the libraries used to develop the scripts. Could you please provide a requirement file with versions? 

An easy way of doing that is using [`pip freeze`](https://pip.pypa.io/en/stable/cli/pip_freeze/) or [`conda list --explicit > requirements.txt`](https://docs.conda.io/projects/conda/en/4.6.0/_downloads/52a95608c49671267e40c689e0bc00ca/conda-cheatsheet.pdf), for example.

Thank you,

Luca

**Comment by antoinefalisse** [comment]
@modenaxe thanks for reporting the issue. Not sure what is happening, I have never seen that before.

Attached, you can find txt files with outputs of `pip freeze`, `conda list --explicit conda_requirements>`, and `pandas.show_versions()`

[conda_requirements.txt](https://github.com/antoinefalisse/predictsim_mtp/files/8406511/conda_requirements.txt)
[pd_versions.txt](https://github.com/antoinefalisse/predictsim_mtp/files/8406512/pd_versions.txt)
[pip_freeze.txt](https://github.com/antoinefalisse/predictsim_mtp/files/8406513/pip_freeze.txt)

Have you tried to create a fresh environment and just install the required packages as described [here](https://github.com/antoinefalisse/predictsim_mtp#install-requirements)?

Thanks

**Comment by modenaxe** [comment]
@antoinefalisse thank you for sharing the files. 
I think the issue is with pandas: `requirements.txt` creates a virtual environment with the latest version (1.4.2) but the code works with `pandas==1.3.3`. The latest version has some issues with the `.append` and `.concat` methods of the dataframes that originate in `guesses.py` and then affect the call to casadi. I haven't done a proper debugging but an easy fix should be using the following to create the environment:
```
conda create -n 3dpredsim pip spyder pandas==1.3.3
```

**Comment by antoinefalisse** [comment]
@modenaxe thanks. I have fixed the issues (see #5). Should be working fine now. Thanks.
